how-to-guides/data-iterator.ipynb

```diff
@@ -0,0 +1,624 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "80e1c34f-37b9-4304-bddd-12e5b2fbcc63",
+   "metadata": {},
+   "source": [
+    "# How to work with data in the form of iterators\n",
+    "\n",
+    "When the data provider for training or validation is an iterator\n",
+    "(infinite or finite with known or unknown size), here are some basic\n",
+    "examples of how to setup trainer or evaluator."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5253db71-301e-4551-9564-f9db49e6ee03",
+   "metadata": {},
+   "source": [
+    "## Infinite iterator for training\n",
+    "\n",
+    "Let’s use an infinite data iterator as training dataflow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "73990ac8-2529-4055-a50a-2d589312e7fd",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1/3 : 1 - 63.862\n",
+      "1/3 : 2 - 64.042\n",
+      "1/3 : 3 - 63.936\n",
+      "1/3 : 4 - 64.141\n",
+      "1/3 : 5 - 64.767\n",
+      "2/3 : 6 - 63.791\n",
+      "2/3 : 7 - 64.565\n",
+      "2/3 : 8 - 63.602\n",
+      "2/3 : 9 - 63.995\n",
+      "2/3 : 10 - 63.943\n",
+      "3/3 : 11 - 63.831\n",
+      "3/3 : 12 - 64.276\n",
+      "3/3 : 13 - 64.148\n",
+      "3/3 : 14 - 63.920\n",
+      "3/3 : 15 - 64.226\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "State:\n",
+       "\titeration: 15\n",
+       "\tepoch: 3\n",
+       "\tepoch_length: 5\n",
+       "\tmax_epochs: 3\n",
+       "\toutput: <class 'NoneType'>\n",
+       "\tbatch: <class 'torch.Tensor'>\n",
+       "\tmetrics: <class 'dict'>\n",
+       "\tdataloader: <class 'generator'>\n",
+       "\tseed: <class 'NoneType'>\n",
+       "\ttimes: <class 'dict'>"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import torch\n",
+    "from ignite.engine import Engine, Events\n",
+    "\n",
+    "torch.manual_seed(12)\n",
+    "\n",
+    "def infinite_iterator(batch_size):\n",
+    "    while True:\n",
+    "        batch = torch.rand(batch_size, 3, 32, 32)\n",
+    "        yield batch\n",
```

Comment:
What about to use `count` from `itertools`?
cause in the next section `Finite iterator with unknown length` we actually do counting with `range(11)`